go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -202,19 +202,22 @@ func runOvnKube(ctx *cli.Context) error {
 		return nil
 	}
 
-	clusterController := ovncluster.NewClusterController(clientset, factory)
 	if master != "" || node != "" {
 		if master != "" {
 			if runtime.GOOS == "windows" {
 				panic("Windows is not supported as master node")
 			}
-			// run the cluster controller to init the master
-			err := clusterController.StartClusterMaster(master)
+			// run the master controller to init the master
+			ovnController := ovn.NewOvnController(clientset, factory)
+			err := ovnController.StartClusterMaster(master)
 			if err != nil {
 				logrus.Errorf(err.Error())
 				panic(err.Error())
 			}
-			ovnController := ovn.NewOvnController(clientset, factory)
+			// add watchers for relevant resources' events
+			// note: the contents of the below function is not in StartClusterMaster()
+			// because master_test.go calls the above function, and it is hard to capture
+			// all the fake commands from bunch of watchers we are adding in the Run()
 			if err := ovnController.Run(); err != nil {
```

Comment:
`ovnController.Run()` and `ovnController.StartClusterMaster()` could be merged?

(FWIW I don't know if the comment is necessary. Having a `Run`/`Start` method that doesn't get called from unit tests is a pretty common pattern in kubernetes and I would have guessed that that was what was going on here even without the comment.)

Reply:
@danwinship 

ovnController.Run() adds watcher for the various resources the ovnkube-master cares about. Some of these watchers calls syncResources() and that function has lots of `ovn-nbctl` calls.

Now, if we merge ovnController.Run() into ovnController.StartClusterMaster(), then in master_test.go in the FakeExecCmds array I will need to include all of those additional calls. It became very tedious.

I would like to keep things as it was before this PR and then tackle merging those two functions in a  separate PR and enhancing master_test.go with those additional calls.

Reply:
oh, I assumed `StartClusterMaster` was split out for the same reason. But I guess it's split out so that the tests can override `TCPLoadBalancerUUID` and `UDPLoadBalancerUUID` before calling it. OK.

Reply:
> (FWIW I don't know if the comment is necessary. Having a `Run`/`Start` method that doesn't get called from unit tests is a pretty common pattern in kubernetes and I would have guessed that that was what was going on here even without the comment.)

Will remove the comment.